src/Installer/redist-installer/targets/BundledTemplates.targets

```diff
@@ -0,0 +1,216 @@
+    <!-- Create a separate project to for template PackageDownloads, as we want to use a task in core-sdk-tasks to collate them, but we
+       can't use those tasks before normal NuGet restore -->
+    <PropertyGroup>
+      <TemplatePackageDownloadProjectContent>
+        <![CDATA[
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>$(TargetFramework)</TargetFramework>
+  </PropertyGroup>
+  <ItemGroup>
+    @(BundledTemplatePackageDownload->'<PackageDownload Include="%(Identity)" Version="%(Version)" />', '
+    ')
+  </ItemGroup>
+</Project>
+]]>
+      </TemplatePackageDownloadProjectContent>
+      <!-- Escape semicolons as %3B in order to avoid being interpreted as line splits in WriteLinesToFile -->
+      <TemplatePackageDownloadProjectContent>$(TemplatePackageDownloadProjectContent.Replace(';', '%3B'))</TemplatePackageDownloadProjectContent>
+      <TemplatePackageDownloadProjectDirectory>$(IntermediateOutputPath)TemplatePackageDownloader\</TemplatePackageDownloadProjectDirectory>
+      <TemplatePackageDownloadProjectPath>$(TemplatePackageDownloadProjectDirectory)TemplatePackageDownloader.csproj</TemplatePackageDownloadProjectPath>
+    </PropertyGroup>
+
+    <MakeDir Directories="$(TemplatePackageDownloadProjectDirectory)" />
+    <WriteLinesToFile Lines="$(TemplatePackageDownloadProjectContent)"
+                      File="$(TemplatePackageDownloadProjectPath)"
+                      Overwrite="True" WriteOnlyWhenDifferent="True" />
+
+    <MSBuild Projects="$(TemplatePackageDownloadProjectPath)"
+             Targets="Restore" />
+  </Target>
+
+  <Target Name="LayoutTemplates"
+        DependsOnTargets="LayoutTemplatesForSDK;LayoutTemplatesForMSI" />
+
+  <Target Name="LayoutTemplatesForSDK" DependsOnTargets="SetupBundledComponents;CalculateTemplatesVersions;DownloadBundledTemplateNupkgs">
+    <ItemGroup Condition="!$(ProductMonikerRid.StartsWith('win'))">
+      <BundledTemplatesWithInstallPaths Remove="Microsoft.Dotnet.Wpf.ProjectTemplates" />
+      <BundledTemplatesWithInstallPaths Remove="Microsoft.Dotnet.WinForms.ProjectTemplates" />
+    </ItemGroup>
+
+    <Copy SourceFiles="%(BundledTemplatesWithInstallPaths.RestoredNupkgPath)"
+          DestinationFolder="$(RedistLayoutPath)templates\%(BundledTemplatesWithInstallPaths.BundledTemplateInstallPath)"
+          Condition="'%(BundledTemplatesWithInstallPaths.TemplateFrameworkVersion)' == '9.0'" />
+
+    <!-- Copy the current Microsoft.DotNet.Common.ItemTemplates and Microsoft.DotNet.Common.ProjectTemplates packages to the layout location. -->
+    <PropertyGroup>
+      <CurrentTemplateInstallPath Condition="'%(BundledTemplatesWithInstallPaths.TemplateFrameworkVersion)' == '9.0'">%(BundledTemplatesWithInstallPaths.BundledTemplateInstallPath)</CurrentTemplateInstallPath>
+    </PropertyGroup>
+    <Copy SourceFiles="%(RepoTemplates.Identity)"
+          DestinationFolder="$(RedistLayoutPath)templates\$(CurrentTemplateInstallPath)" />
```

Comment:
This and a similar copy in the target below were added to use the local output of the SDK build (when running in `-pack`).